queries/dart/highlights.scm

```diff
@@ -111,35 +149,60 @@
 
 ["import" "library" "export"] @include
 
+; Reserved words (cannot be used as identifiers)
+; TODO: "rethrow" @keyword
+
 [
- "abstract"
- ;"assert"
- ;"break"
- "on"
- "class"
- "default"
- "enum"
- "extends"
- ;"final"
- "implements"
- "as"
- "mixin"
- "external"
- "new"
- "return"
- "static"
- "required"
- "var"
- ;"const"
- "async"
- "await"
- ] @keyword
-;TODO:
-; "rethrow" @keyword
+    ; "assert"
+    "class"
+    "enum"
+    "extends"
+    "in"
+    "is"
+    "new"
+    "return"
+    "super"
+    "with"
+] @keyword
+
+[
+    "abstract"
+    "as"
+    "async"
+    "await"
+    "covariant"
+    "deferred"
+    "dynamic"
+    "external"
+    "factory"
+    "Function"
+    "get"
+    "implements"
+    "interface"
+    "library"
+    "operator"
+    "mixin"
+    "part"
+    "set"
+    "show"
+    "static"
+    "typedef"
+] @keyword
```

Comment:
Why having two classes here ?

Reply:
In the upstream `highlights.scm` these are separated [here](https://github.com/UserNobody14/tree-sitter-dart/blob/07fc4595471ba9d3b2038fe3f6e3bb502988684b/queries/highlights.scm#L144). I removed the comments over time by accident. The intent was to separate reserved keywords from builtin identifiers (the reserved ones can't be variable names) the ones below can. I figured at some future date it might be nice to highlight that differently 🤷🏿‍♂️ . I'll re-add the comments